app/web_ui/src/routes/(app)/settings/manage_tools/[project_id]/add_tools/remote_mcp/+page.svelte

```diff
@@ -0,0 +1,252 @@
+<script lang="ts">
+  import AppPage from "../../../../../app_page.svelte"
+  import FormContainer from "$lib/utils/form_container.svelte"
+  import FormElement from "$lib/utils/form_element.svelte"
+  import FormList from "$lib/utils/form_list.svelte"
+  import { client } from "$lib/api_client"
+  import { page } from "$app/stores"
+  import { goto } from "$app/navigation"
+  import { KilnError, createKilnError } from "$lib/utils/error_handlers"
+
+  // Form fields
+  let name = ""
+  let server_url = ""
+  let description = ""
+
+  // Headers as array of key/value pairs
+  interface HeaderPair {
+    key: string
+    value: string
+    placeholder: string | null
+  }
+
+  let headers: HeaderPair[] = []
+
+  // Form state
+  let error: KilnError | null = null
+  let submitting = false
+
+  // Populate fields from parent page state if provided (only if fields are empty)
+  $: if ($page.state) {
+    const state = $page.state as {
```

Comment:
no casting / never use as. Really: never allowed. It leads to ugly breaking runtime errors.

Check if the fields exist and use them if they do. More

```
const state = $page.state || {}
if ("name" in state && typeof state["name"] === "string") {
    name = state.name
}
```